### PR TITLE
added a new pipeline scripts for daily builds

### PR DESCRIPTION
**Pull request by mbiarnes**
added steps for removing npm and yarn files
New groovy scripts for daily build pipelines and prod tag pipeline.
Tested locally.
Please merge together with: https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1074
(these sh scripts are needed)

### CODE REVIEW

File: job-dsls/jobs/DailyBuild_pipeline.groovy [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+def javaToolEnv="KIE_JDK1_8"
+def mvnToolEnv="KIE_MAVEN_3_5_2"
+def mvnHome="${mvnToolEnv}_HOME"
+def kieVersion=Constants.KIE_PREFIX
+def baseBranch=Constants.BRANCH
+def organization=Constants.GITHUB_ORG_UNIT
+def deployDir="deploy-dir"
+def m2Dir="\$HOME/.m2/repository"
+
+String EAP7_DOWNLOAD_URL = "http://download-ipv4.eng.brq.redhat.com/released/JBoss-middleware/eap7/7.2.0/jboss-eap-7.2.0.zip"
```
Comment (winklerm) [inline job-dsls/jobs/DailyBuild_pipeline.groovy]:
```suggestion
String EAP7_DOWNLOAD_URL = "http://download.devel.redhat.com/released/JBoss-middleware/eap7/7.2.0/jboss-eap-7.2.0.zip"
```

Comment (winklerm) [reply job-dsls/jobs/DailyBuild_pipeline.groovy]: The above change does not force using servers in Brno for download in RDU (there is a local proxy as well there).

File: job-dsls/jobs/DailyBuild_pipeline.groovy [hunk 2]
```diff
@@ -0,0 +1,613 @@
...
+                    '${BUILD_LOG, maxLines=750}', subject: 'daily build of ${baseBranch} ', to: 'bsig@redhat.com'
+            }    
+        }        
+        stage('Unpack zip of artifacts to QA Nexus') {
+            steps {
+                withCredentials([usernameColonPassword(credentialsId: 'unpacks-zip-on-qa-nexus', variable: 'kieUnpack')]) {
+                    // unpack zip to QA Nexus
+                    sh 'cd $deployDir\\n' +
+                            'zip -r kiegroup .\\n' +
+                            'curl --silent --upload-file kiegroup.zip -u $kieUnpack -v http://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8081/nexus/service/local/repositories/kieAllBuild-$baseBranch/content-compressed\\n' +
```
Comment (winklerm) [inline job-dsls/jobs/DailyBuild_pipeline.groovy]:
```suggestion
                            'curl --silent --upload-file kiegroup.zip -u $kieUnpack -v http://${LOCAL_NEXUS_IP}:8081/nexus/service/local/repositories/kieAllBuild-$baseBranch/content-compressed\\n' +
```

File: job-dsls/jobs/DailyBuild_pipeline.groovy [hunk 3]
```diff
@@ -0,0 +1,613 @@
...
+
+// +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+// Additional tests
+
+// definition of jbpmTestCoverageMatrix test
+def jbpmTestCoverage='''#!/bin/bash -e
+STAGING_REP=kie-internal-group
+echo "KIE version: $kieVersion"
+# wget the tar.gz sources
+wget -q http://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8081/nexus/content/repositories/kieAllBuild-$baseBranch/org/jbpm/jbpm/$kieVersion/jbpm-$kieVersion-project-sources.tar.gz -O sources.tar.gz
```
Comment (winklerm) [inline job-dsls/jobs/DailyBuild_pipeline.groovy]:
```suggestion
wget -q http://${LOCAL_NEXUS_IP}:8081/nexus/content/repositories/kieAllBuild-$baseBranch/org/jbpm/jbpm/$kieVersion/jbpm-$kieVersion-project-sources.tar.gz -O sources.tar.gz
```

File: job-dsls/jobs/DailyBuild_pipeline.groovy [hunk 4]
```diff
@@ -0,0 +1,613 @@
...
+        }
+    }
+}
+
+// +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+// run additional test: jbpmContainerTestMatrix test
+def jbpmContainerTest='''#!/bin/bash -e
+echo "KIE version $kieVersion"
+# wget the tar.gz sources
+wget -q http://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8081/nexus/content/repositories/kieAllBuild-$baseBranch/org/jbpm/jbpm/$kieVersion/jbpm-$kieVersion-project-sources.tar.gz -O sources.tar.gz
```
Comment (winklerm) [inline job-dsls/jobs/DailyBuild_pipeline.groovy]:
```suggestion
wget -q http://${LOCAL_NEXUS_IP}:8081/nexus/content/repositories/kieAllBuild-$baseBranch/org/jbpm/jbpm/$kieVersion/jbpm-$kieVersion-project-sources.tar.gz -O sources.tar.gz
```

File: job-dsls/jobs/DailyBuild_pipeline.groovy [hunk 5]
```diff
@@ -0,0 +1,613 @@
...
+        }
+    }
+}
+
+// +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+//  run additional test: kieWbTestsMatrix
+def kieWbTest='''#!/bin/bash -e
+echo "KIE version $kieVersion"
+# wget the tar.gz sources
+wget -q http://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8081/nexus/content/repositories/kieAllBuild-$baseBranch/org/kie/kie-wb-distributions/$kieVersion/kie-wb-distributions-$kieVersion-project-sources.tar.gz -O sources.tar.gz
```
Comment (winklerm) [inline job-dsls/jobs/DailyBuild_pipeline.groovy]:
```suggestion
wget -q http://${LOCAL_NEXUS_IP}:8081/nexus/content/repositories/kieAllBuild-$baseBranch/org/kie/kie-wb-distributions/$kieVersion/kie-wb-distributions-$kieVersion-project-sources.tar.gz -O sources.tar.gz
```

File: job-dsls/jobs/DailyBuild_pipeline.groovy [hunk 6]
```diff
@@ -0,0 +1,613 @@
...
+        }
+    }
+}
+
+// +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+//  run additional test: kieServerMatrix
+def kieServerTest='''#!/bin/bash -e
+echo "KIE version $kieVersion"
+# wget the tar.gz sources
+wget -q http://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8081/nexus/content/repositories/kieAllBuild-$baseBranch/org/drools/droolsjbpm-integration/$kieVersion/droolsjbpm-integration-$kieVersion-project-sources.tar.gz -O sources.tar.gz
```
Comment (winklerm) [inline job-dsls/jobs/DailyBuild_pipeline.groovy]:
```suggestion
wget -q http://${LOCAL_NEXUS_IP}:8081/nexus/content/repositories/kieAllBuild-$baseBranch/org/drools/droolsjbpm-integration/$kieVersion/droolsjbpm-integration-$kieVersion-project-sources.tar.gz -O sources.tar.gz
```

File: job-dsls/jobs/DailyBuild_pipeline.groovy [hunk 7]
```diff
@@ -0,0 +1,613 @@
...
+        shell(kieServerTest)
+        maven{
+            mavenInstallation("${mvnVersion}")
+            goals("-B -e -fae -nsu clean verify -P\$container -Pjenkins-pr-builder")
+            rootPOM("kie-server-parent/kie-server-tests/pom.xml")
+            properties("kie.server.testing.kjars.build.settings.xml":"\$SETTINGS_XML_FILE")
+            properties("maven.test.failure.ignore": true)
+            properties("deployment.timeout.millis":"240000")
+            properties("container.startstop.timeout.millis":"240000")
+            properties("eap7.download.url":EAP7_DOWNLOAD_URL)
```
Comment (winklerm) [inline job-dsls/jobs/DailyBuild_pipeline.groovy]: I am really not sure here, but should there be a $ sign before variable name?
```suggestion
            properties("eap7.download.url":"\$EAP7_DOWNLOAD_URL")
```

File: job-dsls/jobs/DailyBuild_pipeline.groovy (hunk 1 above)
Comment (mbiarnes) [reply job-dsls/jobs/DailyBuild_pipeline.groovy]: done

File: job-dsls/jobs/DailyBuild_pipeline.groovy (hunk 2 above)
Comment (mbiarnes) [reply job-dsls/jobs/DailyBuild_pipeline.groovy]: doen

File: job-dsls/jobs/DailyBuild_pipeline.groovy (hunk 3 above)
Comment (mbiarnes) [reply job-dsls/jobs/DailyBuild_pipeline.groovy]: done

File: job-dsls/jobs/DailyBuild_pipeline.groovy (hunk 4 above)
Comment (mbiarnes) [reply job-dsls/jobs/DailyBuild_pipeline.groovy]: done

File: job-dsls/jobs/DailyBuild_pipeline.groovy (hunk 5 above)
Comment (mbiarnes) [reply job-dsls/jobs/DailyBuild_pipeline.groovy]: done

File: job-dsls/jobs/DailyBuild_pipeline.groovy (hunk 6 above)
Comment (mbiarnes) [reply job-dsls/jobs/DailyBuild_pipeline.groovy]: done

File: job-dsls/jobs/DailyBuild_pipeline.groovy (hunk 7 above)
Comment (mbiarnes) [reply job-dsls/jobs/DailyBuild_pipeline.groovy]: I would put it here too  -but it is working like this in our daily builds (see https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/dailyBuild/job/kieServerMatrix-kieAllBuild-master/configure --> invike top level targets --> advanced and you will see here in properties:
eap7.download.url=http://download-ipv4.eng.brq.redhat.com/released/JBoss-middleware/eap7/7.2.0/jboss-eap-7.2.0.zip - so the variable is displayed right here

File: job-dsls/jobs/DailyBuild_pipeline.groovy [hunk 8]
```diff
@@ -0,0 +1,613 @@
...
+def javaToolEnv="KIE_JDK1_8"
+def mvnToolEnv="KIE_MAVEN_3_5_2"
+def mvnHome="${mvnToolEnv}_HOME"
+def kieVersion=Constants.KIE_PREFIX
+def baseBranch=Constants.BRANCH
+def organization=Constants.GITHUB_ORG_UNIT
+def deployDir="deploy-dir"
+def m2Dir="\$HOME/.m2/repository"
+
+String EAP7_DOWNLOAD_URL = "http://download.devel.eng.brq.redhat.com/released/JBoss-middleware/eap7/7.2.0/jboss-eap-7.2.0.zip"
```
Comment (winklerm) [inline job-dsls/jobs/DailyBuild_pipeline.groovy]:
```suggestion
String EAP7_DOWNLOAD_URL = "http://download.devel.redhat.com/released/JBoss-middleware/eap7/7.2.0/jboss-eap-7.2.0.zip"
```

File: job-dsls/jobs/DailyBuild_pipeline.groovy (hunk 7 above)
Comment (winklerm) [reply job-dsls/jobs/DailyBuild_pipeline.groovy]: OK, let's leave it as it is then.

Comment (winklerm) [reply job-dsls/jobs/DailyBuild_pipeline.groovy]: I see, it is a Groovy (JobDSL) variable, so it makes sense now.

File: job-dsls/jobs/DailyBuild_pipeline.groovy (hunk 8 above)
Comment (winklerm) [reply job-dsls/jobs/DailyBuild_pipeline.groovy]: I think this is the last change pending.

Comment (mbiarnes) [reply job-dsls/jobs/DailyBuild_pipeline.groovy]: DONE

**Pull request merged**